File: services/adminservice/internal/service/logo_resolver_test.go
package service

import "testing"

func TestResolvePodLogoURL(t *testing.T) {
	if got := resolvePodLogoURL("pod-gpu-1", "NVIDIA RTX 4090"); got != "/logos/sharemtc-mark.svg" {
		t.Fatalf("expected shared brand mark logo, got %s", got)
	}
	if got := resolvePodLogoURL("custom", "cpu-only"); got != "/logos/sharemtc-mark.svg" {
		t.Fatalf("expected fallback logo, got %s", got)
	}
}

File: services/frontend/src/lib/logoResolver.test.ts
import { describe, expect, it } from "vitest";
import { resolvePodLogoURL, resolveTemplateLogoURL } from "./logoResolver";

describe("logo resolver", () => {
  it("resolves known template logos", () => {
    expect(resolveTemplateLogoURL("fastpanel", "FastPanel")).toBe("/logos/fastpanel.svg");
    expect(resolveTemplateLogoURL("aapanel", "aaPanel")).toBe("/logos/aapanel.svg");
  });

  it("falls back to brand mark for unknown templates", () => {
    expect(resolveTemplateLogoURL("custom", "Custom")).toBe("/logos/sharemtc-mark.svg");
  });

  it("resolves pod logo to shared brand mark", () => {
    expect(resolvePodLogoURL("pod-gpu", "NVIDIA RTX 4090")).toBe("/logos/sharemtc-mark.svg");
  });
});
